### PR TITLE
show set display language action for lang pack extensions

### LINKED ISSUE #154461: Web: Language extensions should not be installable
On the web, we just make NLS work without installing language extensions. But they still appear installable when listed in the extensions view. One idea: show a `Set Language` button, instead of the `Install` button, or similar.

cc @TylerLeonhardt

### PR DESCRIPTION
**Pull request by sandy081**
show set display language action for lang pack extensions

Fixes #154461

### CODE REVIEW

File: src/vs/workbench/contrib/extensions/browser/extensionsActions.ts [hunk 1]
```diff
@@ -264,11 +265,18 @@ export abstract class AbstractInstallAction extends ExtensionAction {
...
-			}
+		if (!this.extension) {
+			return;
+		}
+		if (this.extension.isBuiltin) {
+			return;
+		}
+		if (this.extensionsWorkbenchService.canSetLanguage(this.extension)) {
+			return;
+		}
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/extensions/browser/extensionsActions.ts:276]: nit: maybe combine these if's?

File: src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts [hunk 2]
```diff
@@ -1248,6 +1252,34 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
...
 	}
 
+	canSetLanguage(extension: IExtension): boolean {
+		if (!isWeb) {
+			return false;
+		}
+
+		if (!extension.gallery) {
+			return false;
+		}
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts:1262]: nit: is 2 ifs really more readable than 1?

**Review by TylerLeonhardt — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
@sandy081 does this change work alright for the pseudo language pack?

File: src/vs/workbench/contrib/extensions/browser/extensionsActions.ts (hunk 1 above)
Comment (sandy081) [reply src/vs/workbench/contrib/extensions/browser/extensionsActions.ts:276]: I would like to have separate checks for each condition for readability (I think its personal :) )

**Comment by sandy081** [comment]
Is that a special language pack? May I know how to try it?

**Pull request merged**